src/clients/UBAClient/UBAClientWithRefresh.ts

```diff
@@ -1,29 +1,32 @@
 import assert from "assert";
 import winston from "winston";
 import { BigNumber } from "ethers";
-import { DepositWithBlock, FillWithBlock, RefundRequestWithBlock, UbaFlow } from "../interfaces";
-import { HubPoolClient, SpokePoolClient } from "../clients";
-import { isDefined, sortEventsAscending } from "../utils";
+import { DepositWithBlock, FillWithBlock, RefundRequestWithBlock, UbaFlow } from "../../interfaces";
+import { HubPoolClient, SpokePoolClient } from "..";
+import { isDefined, sortEventsAscending } from "../../utils";
+import { BaseUBAClient, RequestValidReturnType } from "./UBAClientAbstract";
+import { UBAFeeSpokeCalculator } from "../../UBAFeeCalculator";
 
-export class UBAClient {
+export class UBAClientWithRefresh extends BaseUBAClient {
   // @dev chainIdIndices supports indexing members of root bundle proposals submitted to the HubPool.
   //      It must include the complete set of chain IDs ever supported by the HubPool.
   // @dev SpokePoolClients may be a subset of the SpokePools that have been deployed.
   constructor(
-    private readonly chainIdIndices: number[],
+    chainIdIndices: number[],
```

Comment:
Should these not remain at least readonly? I can't see any reason we'd want them to change after instantiation.

Reply:
Fair point! I didn't realize you could make any variable readonly. You learn something new every day